server/src/main/java/org/elasticsearch/index/snapshots/blobstore/SlicedInputStream.java

```diff
@@ -38,16 +42,19 @@ private InputStream nextStream() throws IOException {
         assert initialized == false || currentStream != null;
         initialized = true;
         IOUtils.close(currentStream);
-        if (slice < numSlices) {
-            currentStream = openSlice(slice++);
+        if (nextSlice < numSlices) {
+            currentStream = openSlice(nextSlice++);
```

Comment:
Since we now have a `closed` field, I wonder whether we should check it before open the next slice?

Reply:
This is not necessary because `close()` makes `initialized = true` and `currentStream = null`, which in turn effectively makes `currentStream()` return null, which effectively makes the read functions to always return -1 and never call `nextStream()`.

Still, happy to add `&& closed == false` to the condition if you think it makes it more readable/safer. But it will never truly matter with the current code.

Reply:
Thanks for the explanation. In that case, I think an assertion is justified similar to existing `currentStream != null`?

Reply:
Added an assertion for closed.